### PR TITLE
Handle circular dependency on root package (fix #118)

### LINKED ISSUE #118: Unable to export a package with a circular dependency
Poetry 1.2.0 cannot export a project with a cyclic dependency. I'm unsure if this is an exporter or a Poetry issue, or what would be involved in fixing it, as I have not dug into it myself yet.

```
$ pipx install poetry==1.2.0
$ git clone https://github.com/python-poetry/poetry && cd poetry
$ poetry export -f requirements.txt -o requirements.txt

Dependency walk failed at poetry (>=1.2.0b3,<2.0.0)
```

Ironically, that version constraint comes from this repository itself (`poetry = ^1.2.0b3`).

### PR DESCRIPTION
**Pull request by brechtm**
Resolves: #118

### CODE REVIEW

File: src/poetry_plugin_export/walker.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def get_python_version_region_markers(packages: list[Package]) -> list[BaseMarke
 def get_project_dependency_packages(
     locker: Locker,
     project_requires: list[Dependency],
+    root_package_name: str,
```
Comment (dimbleby) [inline src/poetry_plugin_export/walker.py:53]: these are all `NormalizedName` (from packaging.utils)

Comment (brechtm) [reply src/poetry_plugin_export/walker.py]: Fixed now!

**Comment by brechtm** [comment]
I finally got around to adding a test case. Please review.

File: tests/test_exporter.py [hunk 2]
```diff
@@ -1130,6 +1130,43 @@ def test_exporter_can_export_requirements_txt_with_nested_packages_cyclic(
...
+    poetry.locker.mock_lock_data(  # type: ignore[attr-defined]
+        {
+            "package": [
+                {
+                    "name": "foo",
+                    "version": "1.2.3",
+                    "category": "main",
+                    "optional": False,
+                    "python-versions": "*",
+                    "dependencies": {"my-package": {"version": "1.2.3"}},
```
Comment (radoering) [inline tests/test_exporter.py:1145]:
```suggestion
                    "dependencies": {poetry.package.pretty_name: {"version": "1.2.3"}},
```

This way, the risk of accidentally making the test useless is reduced.

File: tests/test_exporter.py [hunk 3]
```diff
@@ -1130,6 +1130,43 @@ def test_exporter_can_export_requirements_txt_with_nested_packages_cyclic(
     assert content == expected
 
 
+def test_exporter_can_export_requirements_txt_with_circular_dependency(
```
Comment (radoering) [inline tests/test_exporter.py:1133]: maybe

```suggestion
def test_exporter_can_export_requirements_txt_with_circular_root_dependency(
```

At the moment, it's not clear from its name that the focus of the test is a circular dependency _on the root package_.

**Review by radoering — CHANGES_REQUESTED** [review]

**Comment by brechtm** [comment]
Thanks for your suggestions, @ radoering. I have included the changes in my last commit.

**Review by radoering — APPROVED** [review]

**Pull request merged**